sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -84,17 +84,27 @@ object BindReferences extends Logging {
       expression: A,
       input: Seq[Attribute],
```

Comment:
I wonder whether we can push the map construction up one level so that we can amortize its cost across multiple `bindReference` calls.